packages/fether-react/src/assets/theme/shared/styledComponents/DivWindowContent.styles.js

```diff
@@ -3,4 +3,8 @@
 //
 // SPDX-License-Identifier: BSD-3-Clause
 
-export * from './colors';
+import styled from 'styled-components';
```

Comment:
I don't think putting any JS code inside assets/ is a good idea, assets/ are really for static assets, but this JS actually exports React component.

I think we can create a new folder inside src/, called `shared/` or `theme/` or `components/`, and put all shared components inside that folder. Probably stuff like `Health` and `RequireHealthOverlay` could also go inside that folder.

The only problem is that this shared components folder will play a similar role as fether-ui. So instead of putting all above components inside the new shared components folder, I'm also okay with putting everything inside fether-ui.

@ltfschoen If you have any proposals on how to organize UI components, I'd like to hear.

Reply:
I just read through a couple of blogposts and summarised my thoughts below

* BEM

	* Components
		* Containers manage data architecture (state, MobX). Not responsible for
		display UI

	* Blocks
		* Presentational Component Blocks render child Elements.
		Handle UI logic. No dependencies. Receive props and render children/DOM.
		Not connected to application state. No handling of business logic.
		BEM naming convention, Avoid nesting Blocks. Flat folder structure.
		i.e.

```
<Card>
  <Card.Header>
    <Card.Image
    ...
```

	* Elements
		* Smallest UI parts. Handle Modifiers to modify the Elements' styles.
		Not bound to a Block's context.

	* Modifiers
		* Functions passed as props to Eleemnts that alter their CSS properties.
			* i.e. animations
		* Resources:
			* styled-components-modifiers

* Folder Structure

src/

	components/									// SMACS State
		App.js 										// Container Component

		shared/

	blocks/											// SMACS Layout/Modules (i.e. flexbox)
		Card/
			Header/
				Header.js							// Element
				Header.styles.js 			// Modifier (specific)
				index.js
			Image.js								// Element
			Text.js									// Element
			Title.js								// Element
			index.js								// Block Component (Presentational)

		shared/										// Move into Fether UI Library
			GlobalStyles.js
				index.js							// Block + import Modifier (theme)

	elements										// SMACS Modules
		A.js											// Element
		DivStyleLeftIn.js					// Element + import Modifier (animation)
		H1.js											// Element
		Link.js										// Element
		P.js											// Element

		shared/										// Move into Fether UI Library

	modifiers
		shared
			themes										// SMACSS Base/Theme
				GlobalStyles.styles.js  // Modifier (theme)
			animations
				StyleLeftIn.styles.js		// Modifier (animation)

	assets
		images
		fixtures
		sass 												// Legacy Sass

* References
	* https://tech.decisiv.com/structuring-our-styled-components-part-i-2bf21fa64b28
	* https://www.toptal.com/javascript/styled-components-library

Reply:
I've just sent you a link to a file named bem_styled_components.md

Reply:
I like it, but I think i haven't 100% understood it:
- `src/components/` This will hold stateful components, so all our Accounts, App, Tokens, Send and Whitelist components (and sub-components) will go under here, correct?
- `src/blocks` Do you have examples from our project which would go here? I'm not sure I understood this well, but for me, everything in fether-ui, + probably Health, RequireHealthOverlay, Scanner would go inside this folder, correct? In this case do we still keep fether-ui?
- `src/elements` Yes, good idea, stuff like Button, Clickable, go there.
- `src/modifiers` sounds good to me too!

^ is that more/less what you had in mind too?